aws/eks/eks-how-to/build-ubuntu-pro-cluster.rst

```diff
@@ -0,0 +1,103 @@
+Build an Ubuntu Pro K8s cluster
+===============================
+
+Pre-requirements
+~~~~~~~~~~~~~~~~
+
+- Having Packer installed: ``sudo snap install packer``
+- Having eksctl installed: `check here <https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html>`_
+- Having your AWS access key id and secret access key
+- Having your Ubuntu pro token
+
+Packer
+~~~~~~
+
+Get and modify the packer file
+******************************
+
+So first, you need to download this `file <https://github.com/carlosbravoa/ubuntu-aws-examples/blob/main/packer_examples/eks-ubuntu-fips.json>`_. We will use it with Packer to build our ami, so first you need to add your information in the variable section, you can also modify the region of the source ami. Don't forget that our final ami needs to be in the same region as where you want to have your cluster.
```

Comment:
why do we use packer here to create a new image? Can't we just call the `apt install pro` and `pro attach $TOKEN` do via cloud-init before calling the bootstrap.sh script?
That seems to me way easier and ppl don't have to deal with registering images to the new region. Also if we release a new image, ppl don't have to rebuild their image.

Reply:
+1 for cloud-init. Image building will also cause additional costs for people (AMI and Snapshot storage), and add complexity (by needing to use another service, learn another tool, etc.)

Reply:
cloud-init was one of the options on the table, but after several iterations, it was deemed the less sustainable option. @SamirPS can provide further details but the main drawbacks were:
 - it forces the use of launch-templates, where the UA token would need to be in plain text
 - when using fips, the nodes have to reboot

There were other concerns wrt the number of operations required to manage such clusters when using launch templates but I'll let @SamirPS provide further details

Reply:
We use packer, because to enable fips we need to reboot the instance and we are not sure that the node would come back to the cluster. So we can't use cloud-init or launchtemplate for fips